### PR TITLE
Make config load errors more friendly

### LINKED ISSUE #637: Enhancement: More user-friendly error message
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Motivation

I get two kinds of error messages when I exec kvrock2redis like this:
* if the `data-dir` in config file is an unexist path,  it shows `Failed to load config, err: NotFound: `
* if the config file is not found,  it shows `Failed to load config, err: No such file or directory`

I think it need a more accurate error message to distinguish this two cases. 

### Solution

Maybe the simpliest way to solve this problem is to append `+ data_dir` after `s.ToString()`:
https://github.com/apache/incubator-kvrocks/blob/c391953f92fc03b4d22e100a42024a898f3fae47/tools/kvrocks2redis/config.cc#L143

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by torwig**
Enrich error messages that can occur during the loading of the configuration for both `Kvrocks` and `Kvrocks2Redis`.
Minor refactoring of edited source files.
Add a description of default values to `Kvrocks2Redis` configuration file (`cluster-enable` changed to `no` because in the source code, the default value is `false`).

Examples of new error messages
1. Kvrocks
```
Failed to load config. Error: failed to open file '/home/yaroslav/tests/kvrocks/kvrocks3.conf': No such file or directory
Failed to load config. Error: at line #L52: failed to set value of field 'cluster-enabled': argument must be 'yes' or 'no'
Failed to load config. Error: at line #L332: compaction-checker-range is invalid: invalid range format, the range should be between 0 and 24
Failed to load config. Error: at line #L24: malformed config line: config line ends unexpectedly in quoted string
```

2. Kvrocks2Redis
```
Failed to load config. Error: the specified Kvrocks working directory './kvrocks-data/' doesn't exist
Failed to load config. Error: at line #L22: Kvrocks port value should be between 0 and 65535
Failed to load config. Error: failed to open file '/home/yaroslav/tests/kvrocks/kvrocks2redis2.conf': No such file or directory
```

Closes: #637

### CODE REVIEW

**Review by git-hulk — DISMISSED** [review]

File: utils/kvrocks2redis/kvrocks2redis.conf [hunk 1]
```diff
@@ -23,7 +29,7 @@ kvrocks 127.0.0.1 6666
 
 # Enable cluster mode.
 #
-# Default: yes
+# Default: no
 cluster-enable yes
```
Comment (PragmaTwice) [inline utils/kvrocks2redis/kvrocks2redis.conf]: Is there a typo?

Comment (torwig) [reply utils/kvrocks2redis/kvrocks2redis.conf]: @PragmaTwice You mean the default is `no` but the actual value in the config file is `yes`? I was thinking about the change. Should I change it to `no`?

Comment (PragmaTwice) [reply utils/kvrocks2redis/kvrocks2redis.conf]: Yeah but I am not familiar with these components. cc @git-hulk

Comment (git-hulk) [reply utils/kvrocks2redis/kvrocks2redis.conf:33]: This configuration should be meanless to kvrocks2redis

Comment (git-hulk) [reply utils/kvrocks2redis/kvrocks2redis.conf]: Yes, the default value should be `no`

**Review by git-hulk — APPROVED** [review]

File: utils/kvrocks2redis/kvrocks2redis.conf (hunk 1 above)
Comment (torwig) [reply utils/kvrocks2redis/kvrocks2redis.conf]: Changed.

**Comment by torwig** [comment]
It seems like the `cluster-enable` config option isn't used across the `kvrocks2redis` codebase. But I'll double-check it later and do the refactoring if it's true (in another PR).

**Review by PragmaTwice — APPROVED** [review]

**Comment by git-hulk** [comment]
> It seems like the `cluster-enable` config option isn't used across the `kvrocks2redis` codebase. But I'll double-check it later and do the refactoring if it's true (in another PR).

It's used when decoding the key, coz the key should have the slot prefix if the cluster mode was enabled.

**Comment by torwig** [comment]
Thanks all. Merging...

**Pull request merged**